online-shop/src/app/auth/auth.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { User } from '../classes/user';
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { UserDetail } from '../classes/user-detail';
+enum Roles {
+  ADMIN,
+  USER,
+  CUSTOMER,
+  NONE,
+}
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  readonly BASE_URL = 'http://localhost:3000';
+  roles: String[] = [];
```

Comment:
beware these are still public here, try and make them private since you made setters and getters for them